### PR TITLE
Updates activity feed macros structure + adds cfpb_report type

### PR DESCRIPTION
**Pull request by anselmbradford**
Updates activity feed and related macros to use structure in https://github.com/cfpb/cfgov-refresh/issues/597

## Additions

- Added `cfpb_report` activity type to activities feed macro.

## Changes

Moved `activity_snippets` macro from `post-macros.html` to `macros/activity-snippet.html`
  and adds render method.
- Made `activity_snippet` macro private.
- Moved `category_icon` macro from `post-macros.html` to `macros/category-icon.html`
  and adds render method.
- Moved `string_length` macro from `macros.html` to `macros/util/text.html`.

## Testing

- /offices/* and /sub_pages/* should display activities feeds as expected. 

## Review

- @KimberlyMunoz 
- @jimmynotjim 
- @sebworks 
- @kurtw 

## Notes

- Reports should show up when those are available in WP.
- Since it's hard to see the changes to the macros, they are as follows:
  - `cfpb_reports` activity type being added to https://github.com/cfpb/cfgov-refresh/pull/650/files#diff-5fedd58294eea4b46fe2e524f52ad20dR20, 
  - activity_snippet made private (also code comment on L:43 is updated) https://github.com/cfpb/cfgov-refresh/pull/650/files#diff-5fedd58294eea4b46fe2e524f52ad20dR37
  - activity_snippets update to use render method like other macros that spit out HTML templates https://github.com/cfpb/cfgov-refresh/pull/650/files#diff-5fedd58294eea4b46fe2e524f52ad20dR20

### CODE REVIEW

File: _includes/macros/activity-snippets.html [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+   List a bunch of activity snippets when given:
+
+   tag:            A tag to use in the query.
+
+   activity_types: An array of activity types.
+                   Possible values: posts, op-ed, press release, speech, testimony
+
+   ========================================================================== #}
+
+{% macro render(tag, activity_types=['posts', 'op-ed', 'press release', 'speech', 'testimony', 'cfpb_report']) %}
```
Comment (anselmbradford) [inline _includes/macros/activity-snippets.html]: @kurtw please scrutinize the addition of `cfpb_report`. Is that how it needs to be?

**Comment by anselmbradford** [comment]
Forgot to mention, fixes https://github.com/cfpb/cfgov-refresh/issues/649

**Comment by anselmbradford** [comment]
Need to update tests...

File: _includes/macros/activity-snippets.html [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+        {% set header = 'Op-Ed' %}
+        {% set feed = queries[doc_type].search_with_url_arguments(size=quantity,
+            filter_category='Op-Ed', filter_tags=tags) %}
+    {% elif activity_type|lower == 'press release' %}
+        {% set doc_type = 'newsroom' %}
+        {% set icon = category_icon.render(activity_type) %}
+        {% set header = 'Press Release' %}
+        {% set feed = queries[doc_type].search_with_url_arguments(size=quantity,
+            filter_category='Press Release', filter_tags=tags) %}
+    {% elif activity_type|lower == 'speech' %}
```
Comment (sebworks) [inline _includes/macros/activity-snippets.html]: Do we need to have an if condition for `cfpb_report` activity type?

**Comment by sebworks** [comment]
Getting `TemplateNotFound: macros/category_icon.html` error when clicking on Activity Feed link.

File: _includes/macros/activity-snippets.html (hunk 2 above)
Comment (anselmbradford) [inline _includes/macros/activity-snippets.html]: Yes! Added in https://github.com/cfpb/cfgov-refresh/commit/588f1e501da87991b158c0c4ff9732cb605372f2 but please scrutinize, I'm not 100% sure that's how it's supposed to be.

**Comment by anselmbradford** [comment]
> Getting TemplateNotFound: macros/category_icon.html error when clicking on Activity Feed link.

Fixed in https://github.com/cfpb/cfgov-refresh/commit/de2b91a49495688bc13cadd0dc4fa5df9100afd4

File: _includes/macros/activity-snippets.html (hunk 2 above)
Comment (anselmbradford) [inline _includes/macros/activity-snippets.html]: Also, added https://github.com/cfpb/cfgov-refresh/commit/711a8ef669865e8695e0798df319eff3fe552d87 for the icons, but left the icon off for reports as one isn't shown on the mockups.

**Comment by sebworks** [comment]
:+1: , not sure on the cfpb_reports actvity type but I would suggest just creating an issue for it and we can talk with @kurtw on Monday. Merge at will.

**Comment by anselmbradford** [comment]
Opened https://github.com/cfpb/cfgov-refresh/issues/654 for `cfpb_reports` name.

**Pull request merged**